packages/scaffold-ui/src/views/w3m-select-addresses-view/index.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  AccountController,
+  ConnectionController,
+  ModalController,
+  OptionsController,
+  RouterController,
+  type AccountType
+} from '@web3modal/core'
+import { customElement } from '@web3modal/ui'
+import { LitElement, html } from 'lit'
+import { state } from 'lit/decorators.js'
+import { ifDefined } from 'lit/directives/if-defined.js'
+import styles from './styles.js'
+
+@customElement('w3m-select-addresses-view')
+export class W3mSelectAddressesView extends LitElement {
+  public static override styles = styles
+  // -- Members ------------------------------------------- //
+  private readonly metadata = OptionsController.state.metadata
+  public allAccounts: AccountType[] = AccountController.state.allAccounts
+  private selectedAccounts: AccountType[] = AccountController.state.allAccounts
+  private selectAll = true
+  private approved = false
+
+  @state() private isApproving = false
+  constructor() {
+    super()
+    AccountController.subscribeKey('allAccounts', allAccounts => {
+      this.allAccounts = allAccounts
+      this.requestUpdate()
+    })
+  }
+
+  onSelectAll = (event: Event) => {
+    const checked = (event.target as HTMLInputElement).checked
+    this.selectAll = this.selectedAccounts.length === this.allAccounts.length
+    this.allAccounts.forEach(account => {
+      this.onSelect(account, checked)
+    })
+  }
```

Comment:
Can we make these functions `private` and use named function rather than anonymous functions to follow same convention and also not having hoisting issues please? Also we could move them to bottom (the private methods section)
```suggestion
  private onSelectAll(event: Event) {
    const checked = (event.target as HTMLInputElement).checked
    this.selectAll = this.selectedAccounts.length === this.allAccounts.length
    this.allAccounts.forEach(account => {
      this.onSelect(account, checked)
    })
  }
```